client/src/api.ts

```diff
@@ -25,35 +46,144 @@ export const createElection = async (data: {
       electionId: `uuid-${Date.now()}`, // This will be a UUID
     }
   }
-  return await fetchJSON('/elections', {
+  return await fetchJSON(`/api/mailelection`, {
     method: 'post',
     body: JSON.stringify(data),
   })
 }
 
-export const getElections = async () => {
+export const getElections = async (): Promise<{
+  mailElections: MailElection[]
+}> => {
   if (useFakeAPI) {
     await sleep()
-    return {
-      elections: [
-        {
-          electionId: 'asdf',
-        },
-      ],
-    }
+    return { mailElections: [mailElection] }
   }
-  return await fetchJSON('/elections')
+  return await fetchJSON(`/api/mailelection`)
 }
 
 export const getElection = async (
   electionId: string
-): Promise<ElectionsListItem> => {
+): Promise<MailElection> => {
   if (useFakeAPI) {
-    return {
-      id: electionId,
-      name: 'foo',
-      createdAt: new Date().toISOString(),
-    }
+    return mailElection
   }
-  return fetchJSON(`/elections/${electionId}`)
+  return fetchJSON(`/api/mailelection/${electionId}`)
 }
+
+export const approveElection = async (data: {
+  electionId: string
+  userId: string
+}) =>
+  await fetchJSON(`/api/mailelection/${data.electionId}/approve`, {
+    method: 'post',
+    body: JSON.stringify(data),
+  })
+
+export const putElectionDefinition = async ({
+  electionId,
+  election,
+}: {
+  electionId: string
+  election: Election
+}) =>
+  await fetchJSON(`/api/mailelection/${electionId}/definition`, {
+    method: 'put',
+    body: JSON.stringify(election),
+  })
+
+export const getElectionDefinition = async ({
+  electionId,
+}: {
+  electionId: string
+}): Promise<Election> =>
+  await fetchJSON(`/api/mailelection/${electionId}/definition`)
+
+export const putBallotTemplate = async ({
+  ballotStyleId,
+  electionId,
+  file,
+  precinctId,
+}: {
+  ballotStyleId: string
+  electionId: string
+  file: Buffer
+  precinctId: string
+}) =>
+  await fetchJSON(
+    `/api/mailelection/${electionId}/ballot-style/${ballotStyleId}/precinct/${precinctId}/template`,
```

Comment:
I've said it before, but I want to reiterate: we need to be very restrictive about what characters are allowed in a precinct or ballot style ID if we're using them in URLs and filenames.

Reply:
Agreed. How best to enforce?

Reply:
Indeed, though we can also urlencode them, although that is a tad ugly if the names get weird.